scripts/scripts.js

```diff
@@ -730,6 +730,35 @@ export function addFavIcon(href) {
   }
 }
 
+/**
+ * returning <a> from given html element.
+ * this function is to get build <a> from nonlink url.
+ * @param {element} elem
+ * @returns <a>
+ */
+export function getTrueLinks(elem) {
```

Comment:
i would have expected this to be called somethink like `getLink()` i am not sure why it would be called `...links` as far as i can tell this would really ever return a single one.

also i think the signature should return a string or a `URL` instead of the `<a>`.

it also feels like the use of regex is overly complicated and also the `<p>` testing etc. shouldn't be necessary. 

if would have expected this thing to be roughly... (just pseudo code)

```
let link;
const a = elem.querySelector('a');
if (a) link = new URL(a.href);
else {
  try {
    link = new URL(elem.textContent) 
  } catch () {
  
  }
}
return link;
```

Reply:
@davidnuescheler 
The main purpose of the function was to fix a URL text that didn't translated as `<a>` from content to html. That is why I made it to return `<a>`. But it also makes sense to me to return just URL for given element. I think that's more simple.

I was also thinking this function should help any element that considered to contain `<a>` so it treated as it should be. 
Then I found out there are several places using this kind of code:
```
elem.querySelectorAll('a').forEach((a) => {
...
}
```
So, I thought it would be nice to make a function to handle the `elem` to clear all the URL text before go through the logic. That's why the function name was plural. But then, I haven't come up with a good solution to handle all the different use cases. I guess, I was overly thinking about this function.

For testing `<p>` is because there are few different use cases for non-link URL element.

For example, here are two non-link URL contents:
![image](https://user-images.githubusercontent.com/55804872/146926071-b1da7e24-1db8-4aab-a872-8054a5961266.png)

These tables turned into HTML like this:
![image](https://user-images.githubusercontent.com/55804872/146927200-a481f37a-aefd-4984-bfc7-76bbd56bdb51.png)

So as you see when the `elem` is sent to the function, it may or may not contain other text than just the URL string.
Therefore, elem.textContent sometimes doesn't just return pure URL string. (for the example above it would return like `https://www.youtube.com/watch?v=2C9bLzY12LQ&ab_channel=ITSub%EC%9E%87%EC%84%ADHere’s a video from youtube.`

As I tested more I found out when it contains other texts, it seems `<p>` gets appeared otherwise it doesn't.
So, I added the "if statement" to handle right target.
For the same reason, I was doing regex to make sure it always return just URL, in case it contains random text in front or end of the textContent.

If you think it is still overly complicated / engineered, I can simply do elem.textContent , though.

In conclusion, I can change the function name, make it to return just URL than <a>, and take the regex out. Let me know what you think.

Reply:
i think that as word turns a URL into a link pretty much with any addition to the URL eg. a return or a trailing space or really anything, the example of having a caption but no link seems a little bit constructed. ...or let me put it that way, the only situation where i have seen this problem in the wild is where it was a URL by itself.
considering that this is a candidate to eventually be resolved on the server-side once we learn more about its frequency, and that i am little bit on the fence to how useful it actually is vs. just teaching the authors to put proper links in anyway, i would probably opt for the simplest possible solution or even just closing the issue #144 (that i reported :) ) with a "won't fix".